units/src/weight.rs

```diff
@@ -198,6 +198,20 @@ crate::internal_macros::impl_op_for_references! {
 
         fn div(self, rhs: Weight) -> Self::Output { self.to_wu() / rhs.to_wu() }
     }
+    impl ops::Rem<u64> for Weight {
+        type Output = Weight;
+
+        fn rem(self, rhs: u64) -> Self::Output {
+            Weight(self.0 % rhs)
+        }
+    }
+    impl ops::Rem<Weight> for Weight {
+        type Output = Weight;
```

Comment:
This one should output an integer type - i.e., how many times does 300 grams of flour go into 700 grams of flour?

Reply:
done

Reply:
That doesn't sound right. For `Div` it's what you say, but for `Rem` is "if I remove 300g of flour from 700g of flour as many times as I can, how much flour I will have left?

Or simply, if you do the algebraic transformation to invert the function will the types still hold?

if `a / b = d` and `a % b = r` then `d * b + r = a`, so `r` must be of the same type as `a`.